### PR TITLE
Fix username not enriched in new credentials modal

### LINKED ISSUE #4453: Username not enriched in new credentials modal when using OIDC
Username is not enriched in the new credentials modal when using OIDC

197392032-5f61515f-190f-44f1-ad7c-163d0d38761a

### PR DESCRIPTION
**Pull request by eladlachmi**
Resolves #4453

### CODE REVIEW

File: webui/src/pages/auth/users/user/credentials.jsx [hunk 1]
```diff
@@ -44,23 +44,25 @@ const UserCredentialsList = ({ userId, after, onPaginate }) => {
             </>
         );
 
+        const msg = (email) => <span>Create a new Access Key for user <strong>{email}</strong>?</span>;
```
Comment (johnnyaug) [inline webui/src/pages/auth/users/user/credentials.jsx:47]:
```suggestion
        const getMsg = (email) => <span>Create new credentials for user <strong>{email}</strong>?</span>;
```

File: webui/src/lib/components/modals.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                <Button variant={variant} onClick={onConfirm}>Yes</Button>
+                <Button variant="secondary" onClick={onHide}>Cancel</Button>
+            </Modal.Footer>
+        </Modal>
+    );
+};
+
+export const ConfirmationButtonWithContext: FC<ConfirmationButtonWithContextProps> = ({ userId, msg, onConfirm, variant, modalVariant, size, disabled = false, tooltip = null, children }) => {
+    const getUserEmailById = useContext(GetUserEmailByIdContext);
+    const email = getUserEmailById(userId);
```
Comment (johnnyaug) [inline webui/src/lib/components/modals.tsx:55]: Call this only when necessary

**Review by johnnyaug — APPROVED** [review]
Nice! minor comments

**Review by guy-har — APPROVED** [review]
LGTM

File: webui/src/pages/auth/users/user/credentials.jsx (hunk 1 above)
Comment (eladlachmi) [reply webui/src/pages/auth/users/user/credentials.jsx]: Done

File: webui/src/lib/components/modals.tsx (hunk 2 above)
Comment (eladlachmi) [reply webui/src/lib/components/modals.tsx]: Good point. Added `useMemo`.

**Review by ortz — APPROVED** [review]

**Pull request merged**